### PR TITLE
feat: update cnvkit pons

### PR DESCRIPTION
**Pull request by mathiasbio**
## Description

[PR description]

#### Added

- [Description]

#### Changed

- [Description]

#### Fixed

- [Description]

#### Removed

- [Description]

## Documentation

- [ ] N/A
- [ ] Updated Balsamic documentation to reflect the changes as needed for this PR.
    - [Document Name]

### Feature Tests

- [ ] N/A
- [ ] Test [Description]
    - [Screenshot]

### Pipeline Integrity Tests

- **Report deliver (generation of the `.hk` file)**
    - [x] N/A
    - [ ] Verified
- **TGA T/O Workflow**
    - [x] N/A
    - [ ] Verified
- **TGA T/N Workflow**
    - [x] N/A
    - [ ] Verified
- **UMI T/O Workflow**
    - [x] N/A
    - [ ] Verified
- **UMI T/N Workflow**
    - [x] N/A
    - [ ] Verified
- **WGS T/O Workflow**
    - [x] N/A
    - [ ] Verified
- **WGS T/N Workflow**
    - [x] N/A
    - [ ] Verified
- **QC Workflow**
    - [x] N/A
    - [ ] Verified
- **PON Workflow**
    - [x] N/A
    - [ ] Verified

## Clinical Genomics Stockholm

<details>
<summary></summary>

### Documentation

- **Atlas documentation**
    - [x] N/A
    - [ ] Updated: [Link]
- **Web portal for Clinical Genomics**
    - [x] N/A
    - [ ] Updated: [Link]

### Panel of Normal specific criteria

- [ ] The PR includes the addition of a new Panel of Normals
- [ ] The samples have been verified to adhere to the sample selection criteria on [Atlas PoN creation instructions for Balsamic](https://atlas.scilifelab.se/infrastructure/BALSAMIC/balsamic_pon/#procedure)

### User Changes

- [x] N/A
- [ ] This PR affects the output files or results.
    - [ ] User feedback is considered unnecessary because [Justification].
    - [ ] Affected users have been included in the development process and given a chance to provide feedback.

### Infrastructure Changes

- **Stored files in Housekeeper**
    - [x] N/A
    - [ ] Updated: [Link]
- **CG (CLI and delivered/uploaded files)**
    - [x] N/A
    - [ ] Updated: [Link]
- **Servers (configuration files on Hasta)**
    - [x] N/A
    - [ ] Updated: [Link]
- **Scout interface**
    - [x] N/A
    - [ ] Updated: [Link]

</details>

## Checklist

> [!IMPORTANT]  
> Ensure that all checkboxes below are ticked before merging.

### For Developers

- **PR Description**
    - [ ] Provided a comprehensive description of the PR.
    - [ ] Linked relevant user stories or issues to the PR.
- **Documentation**
    - [ ] Verified and updated documentation if necessary.
- **Tests**
    - [ ] Described and tested the functionality addressed in the PR.
    - [ ] Ensured integration of the new code with existing workflows.
    - [ ] Confirmed that meaningful unit tests were added for the changes introduced.
    - [ ] Checked that the PR has successfully passed all relevant code smells and coverage checks.
- **Review**
    - [ ] Addressed and resolved all the feedback provided during the code review process.
    - [ ] Obtained final approval from designated reviewers.

### For Reviewers

- **Code**
    - [ ] Code implements the intended features or fixes the reported issue.
    - [ ] Code follows the project's coding standards and style guide.
- **Documentation**
    - [ ] Pipeline changes are well-documented in the CHANGELOG and relevant documentation.
- **Tests**
    - [ ] The author provided a description of their manual testing, including consideration of edge cases and boundary
      conditions where applicable, with satisfactory results.
- **Review**
    - [ ] Confirmed that the developer has addressed all the comments during the code review.

### CODE REVIEW

File: BALSAMIC/constants/cluster_cache.json [hunk 1]
```diff
@@ -53,6 +53,12 @@
         "mail_type": "FAIL",
         "partition": "core"
     },
+    "expand_short_bedregions": {
+        "n": 1,
+        "time": "00:15:00",
+        "mail_type": "FAIL",
+        "partition": "core"
+    },
```
Comment (ivadym) [inline BALSAMIC/constants/cluster_cache.json:61]: Added by mistake to the cache constants?

```suggestion
```

File: BALSAMIC/models/params.py [hunk 2]
```diff
@@ -194,6 +194,16 @@ class BAMPostProcessingParams(BaseModel):
     manta_max_base_quality: int
 
 
+class BEDPreProcessingParams(BaseModel):
+    """This class defines the params settings used as constants in bed pre-processing rules
```
Comment (ivadym) [inline BALSAMIC/models/params.py:198]:
```suggestion
    """This class defines the params settings used as constants in bed pre-processing rules.
```

File: BALSAMIC/snakemake_rules/variant_calling/somatic_cnv_tumor_normal_tga.rule [hunk 3]
```diff
@@ -70,69 +65,33 @@ rule cnvkit_segment_CNV_research:
...
         sample_id = "TUMOR",
         normal_id = "NORMAL",
-        min_mapq = params.common.min_mapq,
         normal_reference_cnn = cnv_dir + "normal_reference.cnn",
         pon = pon_cnn,
     message:
         "Segmenting genomic regions using CNVkit for {params.case_name}"
     shell:
         """
-# create target and anti-target bed files
```
Comment (ivadym) [inline BALSAMIC/snakemake_rules/variant_calling/somatic_cnv_tumor_normal_tga.rule:80]: Amazing that you moved this part to its own rule 💯

File: BALSAMIC/snakemake_rules/variant_calling/extend_bed.rule [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        baits_bed = config_model.panel.capture_kit,
+        wake_up= result_dir + "start_analysis",
+    output:
+        baits_bed_expanded=Path(cnv_dir + "capture_kit_expanded.bed").as_posix(),
+    benchmark:
+        Path(benchmark_dir,"extend_short_bedregions.tsv").as_posix()
+    singularity:
+        Path(singularity_image,config["bioinfo_tools"].get("pysam") + ".sif").as_posix()
+    params:
+        bedfile_extend = get_script_path("extend_bedfile.py"),
```
Comment (ivadym) [inline BALSAMIC/snakemake_rules/variant_calling/extend_bed.rule:13]:
```suggestion
        bedfile_extend_script = get_script_path("extend_bedfile.py"),
```

File: docs/balsamic_pon.rst [hunk 5]
```diff
@@ -54,6 +54,8 @@ When creating a new PON reference file, the next steps have to be followed:
     /path/analysis/analysis_PON_finish
     /path/analysis/cnv/<panel_name>_CNVkit_PON_reference_<version>.cnn
 
+``Note: The bedfile from the bait-set will be padded in the generation of the PON according to the minimum bed region size set in Balsamic as well as during the analysis with CNVkit, this to avoid CNVkit filtering out short regions.``
```
Comment (ivadym) [inline docs/balsamic_pon.rst:57]: We could use RST specific note notation https://sublime-and-sphinx-guide.readthedocs.io/en/latest/notes_warnings.html

**Review by ivadym — APPROVED** [review]
Looks great, neat implementation 🧙‍♂️ 🪄

File: BALSAMIC/constants/cluster_cache.json (hunk 1 above)
Comment (mathiasbio) [reply BALSAMIC/constants/cluster_cache.json:61]: You betcha! 🤪 Thanks!

File: docs/balsamic_pon.rst (hunk 5 above)
Comment (mathiasbio) [reply docs/balsamic_pon.rst:57]: Good idea, I'll use the warning note to make it more clear!
